internal/controller/warehouses/git.go

```diff
@@ -55,7 +56,16 @@ func (r *reconciler) selectCommits(
 			logger.Debug("found no credentials for git repo")
 		}
 
-		gm, err := r.selectCommitMetaFn(ctx, *s.Git, repoCreds)
+		repoCommitMappings := map[string]string{}
+		if lastFreight != nil {
+			for _, commit := range lastFreight.Commits {
+				repoCommitMappings[commit.RepoURL+"#"+commit.Branch] = commit.ID
+			}
+		}
```

Comment:
Would it be better to build this map just once before the for loop?

Reply:
it sure would! 🤦‍♂️ deemed it too small a fix after a long working day 🙂 Thanks @krancour 
https://github.com/akuity/kargo/pull/1477/commits/304f7ca24d1c8d1cadc16b512a35377b796f9298